src/prop/prop_proof_manager.cpp

```diff
@@ -338,9 +340,70 @@ Node PropPfManager::normalizeAndRegister(TNode clauseNode,
   {
     d_satPm->registerSatAssumptions({normClauseNode});
   }
+  // if proof logging, make the call now
+  if (d_plog != nullptr)
+  {
+    if (!input)
+    {
+      if (d_env.isTheoryProofProducing())
+      {
+        // if theory proof producing, we get the proof to log
+        std::shared_ptr<ProofNode> pfn = d_proof.getProofFor(normClauseNode);
```

Comment:
so if proof logging and doing theory proofs we always do them eagerly... I guess that's how it has to be indeed.

Reply:
Yes, indeed this is a bit of a suboptimal combination, but this makes the behavior consistent.